runelite-client/src/main/java/net/runelite/client/util/ExamineType.java

```diff
@@ -22,12 +22,67 @@
  * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
-package net.runelite.client.plugins.examine;
+package net.runelite.client.util;
+
+import javax.annotation.Nullable;
+import net.runelite.api.ChatMessageType;
+import net.runelite.api.MenuAction;
 
 public enum ExamineType
 {
+	GROUND_ITEM,
 	ITEM,
 	ITEM_BANK_EQ,
 	NPC,
 	OBJECT;
+
+	/**
+	 * Gets the {@link ExamineType} corresponding to the given {@link ChatMessageType}.
+	 *
+	 * @param chatMessageType The chat message type to get an appropriate examine type from
+	 * @return A corresponding examine type object
+	 */
+	@Nullable
+	public static ExamineType of(ChatMessageType chatMessageType)
+	{
+		switch (chatMessageType)
+		{
+			case EXAMINE_ITEM:
+				return ITEM;
+			case EXAMINE_OBJECT:
+				return OBJECT;
+			case EXAMINE_NPC:
+				return NPC;
+			case SERVER:
+				return ITEM_BANK_EQ;
+			default:
+				return null;
+		}
+	}
+
+	/**
+	 * Gets the {@link ExamineType} corresponding to the given {@link MenuAction}.
+	 *
+	 * @param menuAction The menu action type to get an appropriate examine type from
+	 * @return A corresponding examine type object
+	 */
+	@Nullable
+	public static ExamineType of(MenuAction menuAction)
```

Comment:
Maybe naming this MenuActionItemType would be more approriate? After removing the chatbox stuff from here

Reply:
See the comment above